doc/specs/#1235 - Azure cloud shell connector/spec.md

```diff
@@ -0,0 +1,78 @@
+
+The last step - connecting to the terminal - will be done via a websocket connection to allow easier communication between the app and the server. 
+
+The entire feature will be implemented in an isolated manner - i.e. it should have little to no dependency on the Windows Terminal app itself. This will allow the feature to become a plugin/extension once Windows Terminal supports plugins. More specifically, the connector will ascribe to the existing ITerminalConnection interface, making this simply another type of connection that Windows Terminal can make. 
+
+## UI/UX Design
+
+Upon successful implementation, a new profile option will appear for users as illustrated in the picture below (the profile will have its own unique icon when implemented).
+
+![Az Prof](images/azProf.png)
+
+As for the rest of the UI, the implementation will adopt the user's preferences from the Windows Terminal app. 
+
+## Capabilities
+
+### Accessibility
+
+This feature will not impact accessibility of Windows Terminal. 
+
+### Security
+
+Any feature that connects to a network introduces some security risks. However, with proper usage of Azure AD v1.0 and careful storage of tokens received from the server, these risks will be mitigated. 
+
+### Reliability
+
+This feature will not impact reliability of Windows Terminal. 
+
+### Compatibility
+
+With the implementation being mostly decoupled from the Windows Terminal app itself, no existing code/behaviours should break due to this feature. 
+
+### Performance, Power, and Efficiency
+
+This feature will not impact performance, power or efficiency of Windows Terminal.
+
+## Potential Issues
+
+1. This implementation depends on another open source project, [cpprestsdk](https://github.com/Microsoft/cpprestsdk). Thus, any issues with their code will affect this feature. However, given that cpprestsdk is a Microsoft project, we can expect a level of reliability and also solve issues internally if needed.
+2. The proposed authentication endpoint is Azure AD v1.0 instead of Azure AD v2.0 (also known as Microsoft Identity Platform). Azure AD v1.0 is still supported for now, but there is a risk of it becoming deprecated at some point in the future. However, given that it is once again another Microsoft-owned project, we can request support for it through an internal channel. In the worst case, our implementation can switch to Microsoft Identity Platform (which would only requires some minor edits to the http requests). 

```

Comment:
Did you find any information on the timeline for v2.0 to support the rest of the authentication (and therefore to be a good candidate for switching)?

Reply:
The [docs](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-device-code) simply say "this support will be enabled soon". Would it help if I contact the team to ask for a more detailed estimate on when?

Reply:
It might be nice to send a mail and try to find out their timeline, but it's not fundamental to us moving forward.